cpp/cmake/Modules/ConfigureArrow.cmake

```diff
@@ -38,14 +46,14 @@ endif(ARROW_CONFIG)
 unset(PARALLEL_BUILD)            
 if($ENV{TRAVIS})
```

Comment:
We don't use Travis any more, I sugggest we change this to look for `JENKINS_HOME` env var as that is set in all Jenkins environments.

Reply:
Also not sure we need this as in Jenkins we do parallel builds, will let the team decide if you want to keep this.

If we switch it will slow things down.

@kkraus14 @harrism @raydouglass

Reply:
Then clean up the lines below referring to Travis as well

Reply:
Will deal with this in another issue